vendor/tektelic/encoder_home_sensor.js

```diff
@@ -0,0 +1,700 @@
+    check_encode("moisture_caliberation_dry",
+        function (value) {
+            var converted = [0x5D | 0x80,
+                value & 0xff];
+            ret = ret.concat(converted);
+        },
+        function () {
+            ret = ret.concat([0x5D])
+        }
+    );
+    check_encode("write_to_flash",
+        function (value) {
+            var converted = [0x70 | 0x80,
+                ((value.app_configuration & 0x1) << 5) |
+                ((value.lora_configuration & 0x1) << 6),
+                ((value.restart_sensor & 0x1) << 0)];
+            ret = ret.concat(converted);
+        },
+        function () {
+            ret = ret.concat([0x70])
+        }
+    );
+    check_encode("firmware_version",
+        function (value) {
+        },
+        function () {
+            ret = ret.concat(ret, [0x71])
+        }
+    );
+    check_encode("configuration_factory_reset",
+        function (value) {
+            var converted = [0x72 | 0x80,
+                value & 0xff];
+            ret = ret.concat(converted);
+        },
+        function () {
+            ret = ret.concat([0x72])
+        }
+    );
+    check_encode("payload",
+        function (value) {
+            var converted = base64ToArray(value);
+            ret = ret.concat(converted);
+        },
+        function () {
+        }
+    );
+    function check_encode(prop_name, do_write, do_read) {
+        if (data.hasOwnProperty(prop_name)) {
+            var obj = data[prop_name];
+            if (obj.hasOwnProperty("access")) {
+                var access_value = obj.access;
+                if (access_value == "write") {
+                    do_write(obj.value);
+                } else if (access_value == "read") {
+                    do_read();
+                }
+            } else if (obj.hasOwnProperty("value")) {
+                do_write(obj.value);
+            }
+        }
+    }
+
+    function atob(input) {
+        var chars = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/=';
+        var str = String(input).replace(/[=]+$/, ''); // #31: ExtendScript bad parse of /=
+        if (str.length % 4 === 1) {
+            throw new InvalidCharacterError("'atob' failed: The string to be decoded is not correctly encoded.");
+        }
+        for (
+            var bc = 0, bs, buffer, idx = 0, output = '';
+            buffer = str.charAt(idx++);
+            ~buffer && (bs = bc % 4 ? bs * 64 + buffer : buffer, bc++ % 4) ? output += String.fromCharCode(255 & bs >> (-2 * bc & 6)) : 0
+        ) {
+            buffer = chars.indexOf(buffer);
+        }
+        return output;
+    }
+
+    function base64ToArray(base64) {
+        var binary_string = atob(base64);
+        var len = binary_string.length;
+        var result = [];
+        for (var i = 0; i < len; i++) {
+            result.push(binary_string.charCodeAt(i));
+        }
+        return result;
+    }
+
+    return {
+        'bytes': ret,
+        'port': port
+    };
```

Comment:
This is okay, the function signature is wrong

Reply:
Didn't manage to update the payload formatters. Will start a discussion about this and update later.